app/code/Magento/Backend/Test/Unit/App/Area/FrontNameResolverTest.php

```diff
@@ -233,6 +233,20 @@ public static function hostsDataProvider(): array
                 'customAdminUrl' => '',
                 'expectedValue' => true
             ],
+            'visitingStorefrontWithAdminSubdomain' => [
+                'url' => 'https://magento.loc',
+                'host' => 'magento.loc',
+                'useCustomAdminUrl' => '1',
+                'customAdminUrl' => 'https://admin.magento.loc',
+                'expectedValue' => false
+            ],
+            'visitingAdminInSubdomain' => [
+                'url' => 'https://magento.loc',
+                'host' => 'admin.magento.loc',
+                'useCustomAdminUrl' => '1',
+                'customAdminUrl' => 'https://admin.magento.loc',
+                'expectedValue' => true
```

Comment:
@zapotocnylubos @lbajsarowicz, shouldn't it be a URL ending with a slash like this? (I didn't test it)
Could you check what the actual value you're having here?
```suggestion
                'url' => 'https://magento.loc/',
                'host' => 'magento.loc',
                'useCustomAdminUrl' => '1',
                'customAdminUrl' => 'https://admin.magento.loc/',
                'expectedValue' => false
            ],
            'visitingAdminInSubdomain' => [
                'url' => 'https://magento.loc/',
                'host' => 'admin.magento.loc',
                'useCustomAdminUrl' => '1',
                'customAdminUrl' => 'https://admin.magento.loc/',
                'expectedValue' => true
```

Reply:
Just wondering why we have a "starts with" login for the admin hostname. Shouldn't it be fully compared?
In theory, we can proxy any domain to something like

`admin.m2website.com` to `admin.m2website.com.myowndomain.com`, and it will pass. I'm not sure if that could cause any issues directly, but might lead to finding additional vulnerabilities based on this behavior.

What do you think?